From 730e1367273e57ac047d734f2d45c4976f36f8c1 Mon Sep 17 00:00:00 2001
From: Kai Kang <kai.kang@windriver.com>
Date: Tue, 9 Feb 2021 15:36:09 +0800
Subject: [PATCH 2/2] greeter.c: disallow empty new password

Disallow empty new password when user is required to update it.

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
 src/greeter.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/greeter.c b/src/greeter.c
index 2208711..541197f 100644
--- a/src/greeter.c
+++ b/src/greeter.c
@@ -237,6 +237,10 @@ static void on_entry_activate(GtkEntry* entry)
 		if (pass_expired) {
 			if (!new_pass) {
 				new_pass = g_strdup(gtk_entry_get_text(entry));
+				if (strlen(new_pass) == 0) {
+					new_pass = NULL;
+					gtk_label_set_text((GtkLabel *)info, _("Empty password is not allowed."));
+				}
 				switch_to_input_passwd();
 			} else {
 				tmp = g_strdup(gtk_entry_get_text(entry));
@@ -253,7 +257,7 @@ static void on_entry_activate(GtkEntry* entry)
 						gtk_label_set_text((GtkLabel *)info, _("Maximum number of failed update password attempts exceeded."));
 						switch_to_input_user();
 					}
-				} else if (!strcmp(pass, g_base64_encode((guchar*)new_pass, strlen(new_pass) + 1))) {
+				} else if (pass && !strcmp(pass, g_base64_encode((guchar*)new_pass, strlen(new_pass) + 1))) {
 					// if new password is same as old one
 					g_free(new_pass);
 					new_pass = NULL;
-- 
2.25.1

